src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs

```diff
@@ -422,6 +439,29 @@ private static RecursivePatternSyntax RecursivePattern(SubpatternSyntax subpatte
                 commonReceiver = GetInnermostReceiver(left, lastName, static (identifierName, lastName) => identifierName != lastName);
             }
 
+            var iSymbol = model.GetSymbolInfo(leftNames[^1]).Symbol;
+            if (iSymbol != null && iSymbol.Equals(model.GetSymbolInfo(rightNames[^1]).Symbol))
```

Comment:
no clue why this chekc is happening.  code needs docs.

Reply:
Same as before, I removed it since it wasn't necessary to do this.